`varLib.instancer` fails with cvar table and certain options

I can’t figure out what’s wrong here.

I have a VF exported straight from Glyphs. It has three axes, `wght`, `ital`, `YTDE`.  Depending on the command line options, `varLib.instancer` fails.

[SudoUIVariable.ttf.zip](https://github.com/fonttools/fonttools/files/14359607/SudoUIVariable.ttf.zip)

This works:

```
fonttools varLib.instancer SudoUIVariable.ttf wght=400:400:700 ital=0:1
```

This fails:

```
fonttools varLib.instancer SudoUIVariable.ttf wght=400:400:700 ital=0
```

It seems the problem is related to the `cvar` table?

```
Restricting axes: {'wght': (400, 400, 700), 'ital': (0, 0, 0)}
Loading variable font
Restricted limits: {'wght': (400, 400, 700), 'ital': (0, 0, 0)}
Normalized limits: {'wght': (0, 0, 1, 200, 300), 'ital': (0, 0, 0, 0, 1)}
Instantiating glyf/gvar tables
Instantiating cvt/cvar tables
Instantiating GDEF and GPOS tables
Instantiating FeatureVariations of GSUB table
Instantiating avar table
Instantiating STAT table
Instantiating fvar table
Pruning name table
Saving partial variable font SudoUIVariable-partial.ttf
Traceback (most recent call last):
  File "venv/bin/fonttools", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "venv/lib/python3.12/site-packages/fontTools/__main__.py", line 31, in main
    runpy.run_module(mod, run_name="__main__")
  File "<frozen runpy>", line 229, in run_module
  File "<frozen runpy>", line 88, in _run_code
  File "venv/lib/python3.12/site-packages/fontTools/varLib/instancer/__main__.py", line 5, in <module>
    sys.exit(main())
             ^^^^^^
  File "venv/lib/python3.12/site-packages/fontTools/varLib/instancer/__init__.py", line 1568, in main
    varfont.save(outfile)
  File "venv/lib/python3.12/site-packages/fontTools/ttLib/ttFont.py", line 211, in save
    writer_reordersTables = self._save(tmp)
                            ^^^^^^^^^^^^^^^
  File "venv/lib/python3.12/site-packages/fontTools/ttLib/ttFont.py", line 258, in _save
    self._writeTable(tag, writer, done, tableCache)
  File "venv/lib/python3.12/site-packages/fontTools/ttLib/ttFont.py", line 708, in _writeTable
    tabledata = self.getTableData(tag)
                ^^^^^^^^^^^^^^^^^^^^^^
  File "venv/lib/python3.12/site-packages/fontTools/ttLib/ttFont.py", line 730, in getTableData
    return self.tables[tag].compile(self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "venv/lib/python3.12/site-packages/fontTools/ttLib/tables/_c_v_a_r.py", line 35, in compile
    tupleVariationCount, tuples, data = compileTupleVariationStore(
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "venv/lib/python3.12/site-packages/fontTools/ttLib/tables/TupleVariation.py", line 698, in compileTupleVariationStore
    thisTuple, thisData = v.compile(axisTags, sharedTupleIndices, pointData=p)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "venv/lib/python3.12/site-packages/fontTools/ttLib/tables/TupleVariation.py", line 131, in compile
    assert set(self.axes.keys()) <= set(axisTags), (
AssertionError: ('Unknown axis tag found.', dict_keys(['wght', 'ital']), ['wght', 'YTDE'])
```

--- a/Lib/fontTools/varLib/instancer/__init__.py
+++ b/Lib/fontTools/varLib/instancer/__init__.py
@@ -531,9 +531,13 @@ def changeTupleVariationsAxisLimits(variations, axisLimits):
 def changeTupleVariationAxisLimit(var, axisTag, axisLimit):
     assert isinstance(axisLimit, NormalizedAxisTripleAndDistances)
 
-    # Skip when current axis is missing (i.e. doesn't participate),
+    # Skip when current axis is missing or peaks at 0 (i.e. doesn't participate)
     lower, peak, upper = var.axes.get(axisTag, (-1, 0, 1))
     if peak == 0:
+        # explicitly defined, no-op axes can be omitted
+        # https://github.com/fonttools/fonttools/issues/3453
+        if axisTag in var.axes:
+            del var.axes[axisTag]
         return [var]
     # Drop if the var 'tent' isn't well-formed
     if not (lower <= peak <= upper) or (lower < 0 and upper > 0):

--- a/Tests/varLib/instancer/instancer_test.py
+++ b/Tests/varLib/instancer/instancer_test.py
@@ -2015,6 +2015,19 @@ class LimitTupleVariationAxisRangesTest:
                 0.5,
                 [TupleVariation({"wght": (1.0, 1.0, 1.0)}, [100, 100])],
             ),
+            # test case from https://github.com/fonttools/fonttools/issues/3453
+            (
+                TupleVariation(
+                    {
+                        "wght": (0.0, 1.0, 1.0),
+                        "ital": (0.0, 0.0, 1.0),  # no-op axis gets dropped
+                    },
+                    [100, 100],
+                ),
+                "ital",
+                0.0,
+                [TupleVariation({"wght": (0.0, 1.0, 1.0)}, [100, 100])],
+            ),
         ],
     )
     def test_positive_var(self, var, axisTag, newMax, expected):

